examples/CMSapp/CMSapp.cpp

```diff
@@ -0,0 +1,266 @@
+#include <G4ProductionCutsTable.hh>
+#include <G4RegionStore.hh>
+
+#include <G4Gamma.hh>
+#include <G4Electron.hh>
+#include <G4Positron.hh>
+#include <G4Proton.hh>
+#include <G4ParticleTable.hh>
+
+#include <G4HepEmData.hh>
+#include <G4HepEmState.hh>
+#include <G4HepEmStateInit.hh>
+#include <G4HepEmParameters.hh>
+#include <G4HepEmMatCutData.hh>
+
+#include <VecGeom/base/Stopwatch.h>
+#include <VecGeom/management/GeoManager.h>
+#include <VecGeom/management/BVHManager.h>
+#include <VecGeom/gdml/Frontend.h>
+#include <VecGeom/management/CudaManager.h>
+
+#include <CopCore/SystemOfUnits.h>
+#include <AdePT/ArgParser.h>
+
+const G4VPhysicalVolume *InitGeant4(std::string gdml_file)
+{
+  // Read the geometry, regions and cuts from the GDML file
+  std::cout << "reading " << gdml_file << " transiently on CPU for Geant4 ...\n"; 
+  G4GDMLParser parser;
+  parser.Read(gdml_file, false); // turn off schema checker
+  G4VPhysicalVolume *world = parser.GetWorldVolume();
+  
+  if (world == nullptr) {
+    std::cerr << "CMSapp: World volume not set properly check your setup selection criteria or GDML input!\n";
+    return world;
+  }
+
+  // - PHYSICS
+  // --- Create particles that have secondary production threshold.
+  G4Gamma::Gamma();
+  G4Electron::Electron();
+  G4Positron::Positron();
+  G4Proton::Proton();
+  G4ParticleTable *partTable = G4ParticleTable::GetParticleTable();
+  partTable->SetReadiness();
+
+  // - REGIONS
+  if (world->GetLogicalVolume()->GetRegion() == nullptr) {
+    // Add default region if none available
+    auto defaultRegion = new G4Region("DefaultRegionForTheWorld");  // deleted by store
+    defaultRegion->AddRootLogicalVolume(world->GetLogicalVolume());
+    defaultRegion->SetProductionCuts(G4ProductionCutsTable::GetProductionCutsTable()
+      ->GetDefaultProductionCuts());
+  }
+  for (auto region : *G4RegionStore::GetInstance())
+    region->UsedInMassGeometry(true); // make sure all regions are marked as used
+
+  // - UPDATE COUPLES
+  std::cout << "updating material-cut couples based on " << G4RegionStore::GetInstance()->size() << " regions ...\n";
+  G4ProductionCutsTable *theCoupleTable = G4ProductionCutsTable::GetProductionCutsTable();
+  theCoupleTable->UpdateCoupleTable(world);
+  return world;
+}
+
+const vecgeom::cxx::VPlacedVolume *InitVecGeom(std::string gdml_file, int cache_depth)
+{
+  // Import the gdml file into VecGeom
+  vecgeom::GeoManager::Instance().SetTransformationCacheDepth(cache_depth);
+  vgdml::Parser vgdmlParser;
+  auto middleWare = vgdmlParser.Load(gdml_file.c_str(), false, copcore::units::mm);
+  if (middleWare == nullptr)
+  {
+    std::cerr << "Failed to read geometry from GDML file '" << gdml_file << "'" << std::endl;
+    return nullptr;
+  }
+
+  const vecgeom::VPlacedVolume *world = vecgeom::GeoManager::Instance().GetWorld();
+  if (world == nullptr) {
+    std::cerr << "GeoManager world volume is nullptr" << std::endl;
+    return nullptr;
+  }
```

Comment:
one of the two `if` statements is not correctly formatted, please run `clang-format` on newly added code